firmware/tracing_firmware/main.rs

```diff
@@ -0,0 +1,178 @@
+#![no_std]
+#![no_main]
+
+use core::arch::{asm, global_asm};
+
+use miralis_abi::{failure, log, setup_binary, success};
+
+setup_binary!(main);
+
+fn enable_mcycle_in_smode() {
+    unsafe {
+        let mcounteren: u32;
+        asm!("csrr {}, mcounteren", out(reg) mcounteren);
+        asm!("csrw mcounteren, {}", in(reg) mcounteren | 1);
+    }
+}
+
+fn main() -> ! {
+    unsafe {
+        asm!("la sp, 0x80700000");
+    }
```

Comment:
You are trying to setup the stack of the firmware here right?
The `setup_binary!` macro already creates a proper stack, so this should not be needed, at least not for the firmware.
The OS will need its own stack though.